lib/processPbrMetallicRoughness.js

```diff
@@ -0,0 +1,774 @@
+
+            techniqueParameters.joint = {
+                semantic: 'JOINTS_0',
+                type: webGLConstant
+            };
+            techniqueAttributes.a_weight = 'weight';
+            techniqueParameters.weight = {
+                semantic: 'WEIGHTS_0',
+                type: webGLConstant
+            };
+
+            vertexShader += 'attribute ' + attributeType + ' a_joint;\n';
+            vertexShader += 'attribute ' + attributeType + ' a_weight;\n';
+        }
+
+        vertexShader += 'void main(void) {\n';
+        vertexShader += vertexShaderMain;
+        vertexShader += '}\n';
+
+        // Fragment shader lighting
+        fragmentShader += 'const float M_PI = 3.141592653589793;\n';
+
+        var lambertianDiffuse = '';
+        lambertianDiffuse += 'vec3 lambertianDiffuse(vec3 baseColor) {\n';
+        lambertianDiffuse += '  return baseColor / M_PI;\n';
+        lambertianDiffuse += '}\n\n';
+
+        var fresnelSchlick2 = '';
+        fresnelSchlick2 += 'vec3 fresnelSchlick2(vec3 f0, vec3 f90, float VdotH) {\n';
+        fresnelSchlick2 += '  return f0 + (f90 - f0) * pow(clamp(1.0 - VdotH, 0.0, 1.0), 5.0);\n';
+        fresnelSchlick2 += '}\n\n';
+
+        var fresnelSchlick = '';
+        fresnelSchlick += 'vec3 fresnelSchlick(float metalness, float VdotH) {\n';
+        fresnelSchlick += '  return metalness + (vec3(1.0) - metalness) * pow(1.0 - VdotH, 5.0);\n';
+        fresnelSchlick += '}\n\n';
+
+        var smithVisibilityG1 = '';
+        smithVisibilityG1 += 'float smithVisibilityG1(float NdotV, float roughness) {\n';
+        smithVisibilityG1 += '  float k = (roughness + 1.0) * (roughness + 1.0) / 8.0;\n';
+        smithVisibilityG1 += '  return NdotV / (NdotV * (1.0 - k) + k);\n';
+        smithVisibilityG1 += '}\n\n';
+
+        var smithVisibilityGGX = '';
+        smithVisibilityGGX += 'float smithVisibilityGGX(float roughness, float NdotL, float NdotV) {\n';
+        smithVisibilityGGX += '  return smithVisibilityG1(NdotL, roughness) * smithVisibilityG1(NdotV, roughness);\n';
+        smithVisibilityGGX += '}\n\n';
+
+        var GGX = '';
+        GGX += 'float GGX(float roughness, float NdotH) {\n';
+        GGX += '  float roughnessSquared = roughness * roughness;\n';
+        GGX += '  float f = (NdotH * roughnessSquared - NdotH) * NdotH + 1.0;\n';
+        GGX += '  return roughnessSquared / (M_PI * f * f);\n';
+        GGX += '}\n\n';
+
+        fragmentShader += lambertianDiffuse + fresnelSchlick2 + fresnelSchlick + smithVisibilityG1 + smithVisibilityGGX + GGX;
+
+        var fragmentShaderMain = '';
+        fragmentShaderMain += 'void main(void) {\n';
+
+        // Add normal mapping to fragment shader
+        if (hasNormals) {
+            fragmentShaderMain += '  vec3 ng = normalize(v_normal);\n';
+            if (defined(parameterValues.normalTexture) && (hasTangents || frameState.context._standardDerivatives)) {
```

Comment:
Standard derivatives will need to handled differently since we don't have this info available at conversion time. Check if there is a `#define` associated with the `GL_OES_standard_derivatives` extension, and add that in here. Check out https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Shaders/ShadowVolumeFS.glsl as an example.